pkg/jobs/ingeststopped/ingesting_checker.go

```diff
@@ -0,0 +1,120 @@
+// Copyright 2023 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package ingeststopped
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/cockroachdb/cockroach/pkg/jobs"
+	"github.com/cockroachdb/cockroach/pkg/jobs/jobspb"
+	"github.com/cockroachdb/cockroach/pkg/sql"
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog/catpb"
+	"github.com/cockroachdb/cockroach/pkg/sql/execinfrapb"
+	"github.com/cockroachdb/cockroach/pkg/sql/physicalplan"
+	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
+	"github.com/cockroachdb/cockroach/pkg/sql/types"
+	"github.com/cockroachdb/cockroach/pkg/util/log"
+	"github.com/cockroachdb/cockroach/pkg/util/retry"
+	"github.com/cockroachdb/cockroach/pkg/util/timeutil"
+	"github.com/cockroachdb/cockroach/pkg/util/tracing"
+	"github.com/cockroachdb/errors"
+)
+
+// WaitForNoIngestingNodes contacts all nodes in the cluster and verifies that
+// they are not ingesting data for the given job ID, and continues to try to do
+// so for the specified duration until it succeeds or times out.
+func WaitForNoIngestingNodes(
+	ctx context.Context, execCtx sql.JobExecContext, job *jobs.Job, maxWait time.Duration,
+) error {
+	ctx, sp := tracing.ChildSpan(ctx, "ingeststopped")
+	defer sp.Finish()
+
+	retries := retry.StartWithCtx(ctx, retry.Options{
+		InitialBackoff: time.Second,
+		MaxBackoff:     time.Second * 10,
+	})
+
+	started := timeutil.Now()
+	const maxIngestingCheckDelay = time.Minute * 5
+
+	var lastStatusUpdate time.Time
+	const statusUpdateFrequency = time.Second * 30
+
+	for retries.Next() {
+		err := checkAllNodesForIngestingJob(ctx, execCtx, job.ID())
+		if err == nil {
+			break
+		}
+		log.Infof(ctx, "failed to verify job no longer importing on all nodes: %+v", err)
+
+		if timeutil.Since(started) > maxWait {
+			return err
+		}
+
+		if timeutil.Since(lastStatusUpdate) > statusUpdateFrequency {
+			if statusErr := job.NoTxn().RunningStatus(ctx, func(_ context.Context, _ jobspb.Details) (jobs.RunningStatus, error) {
+				return jobs.RunningStatus(fmt.Sprintf("waiting for all nodes to finish writing before reverting: %s", err)), nil
+			}); statusErr != nil {
+				log.Warningf(ctx, "failed to update running status of job %d: %s", errors.Safe(job.ID()), statusErr)
```

Comment:
[question] Did you find errors.Safe necessary here? I thought we implemented SafeValue on JobID's

[nit] We don't log the JobID in the Infof call above but do here. We could make them consistent.  In the only production caller of this code, I believe we'll have the job ID in the log tags.